cmd/heimdall-token-analytics/server/auth.go

```diff
@@ -81,6 +164,20 @@ func NIP42AuthMiddleware() gin.HandlerFunc {
 			ctx.Next()
 			return
 		}
+		if strings.HasPrefix(token, xcomAuthScheme+" ") {
+			claims, err := authValidateXComToken(token, xcomSecretKey)
```

Comment:
For validation you need only public key, not a secret one. Btw I dont see creation of those tokens, do FE issue them itself?